gcloud-java-compute/src/main/java/com/google/gcloud/compute/Disk.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.gcloud.compute.Compute.DiskOption;
+import com.google.gcloud.compute.Compute.OperationOption;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google Compute Engine persistent disk. A disk can be used as primary storage for your virtual
+ * machine instances. Objects of this class are immutable. To get a {@code Disk} object with the
+ * most recent information use {@link #reload}. {@code Disk} adds a layer of service-related
+ * functionality over {@link DiskInfo}.
+ *
+ * @see <a href="https://cloud.google.com/compute/docs/disks/">Block Storage</a>
+ */
+public class Disk extends DiskInfo {
+
+  private static final long serialVersionUID = 7234747955588262204L;
+
+  private final ComputeOptions options;
+  private transient Compute compute;
+
+  /**
+   * A builder for {@code Disk} objects.
+   */
+  public static class Builder extends DiskInfo.Builder {
+
+    private final Compute compute;
+    private final DiskInfo.BuilderImpl infoBuilder;
+
+    Builder(Compute compute, DiskId diskId, DiskConfiguration diskConfiguration) {
+      this.compute = compute;
+      this.infoBuilder = new DiskInfo.BuilderImpl(diskId, diskConfiguration);
+      this.infoBuilder.diskId(diskId);
```

Comment:
Is this line and the line below unnecessary because `diskId` and `diskConfiguration` are constructor arguments?